lib/request-handler.js

```diff
@@ -78,8 +77,8 @@ RequestHandler.prototype.iterateThroughHosts = function (iterator, callback) {
       return callback(new errors.NoHostAvailableError(this.triedHosts));
     }
     host = item.value;
-    // set the distance relative to the client first, using the load balancing policy
-    var distance = host.getDistance(this.loadBalancingPolicy);
+    // set the distance relative to the client first
+    var distance = this.profileManager.getDistance(host);
```

Comment:
According to the design doc, all load balancing policies should be used to determine the (closest) distance.
If a execution profile uses a lbp that identifies that a host is `remote` / `ignored` but other lbp, in a different exec profile, identifies the same host as `local`. The host is considered as `local` (having a pool according to the distance).